<div style="background:linear-gradient(135deg,#7a3d00 0%,#b45309 55%,#d97706 100%);border-radius:18px;padding:34px 30px;color:#fff;font-family:Inter,Segoe UI,sans-serif">
  <div style="font-size:12px;letter-spacing:3px;color:#ffe9c7;font-weight:700;text-transform:uppercase">Chapter 8 · Solutions</div>
  <div style="font-size:36px;font-weight:900;line-height:1.1;margin:10px 0 6px">Practice Challenges, Worked Answers ✅</div>
  <div style="font-size:15px;color:#ffe6cc;max-width:700px;line-height:1.6">Full solutions to the five "Measures of Central Tendency" challenges. Try them yourself first, then compare.</div>
  <div style="margin-top:16px;font-size:13px;color:#ffe2bf">Statistics, Data Science and AI: A Visual Handbook · John Fisher · 2026</div>
</div>

## ⚙️ Setup

In [1]:
import numpy as np
import pandas as pd
print("✅ Ready.")

✅ Ready.


<div style="background:#fef3e2;border-left:5px solid #d97706;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#0891b2;letter-spacing:1px">CHALLENGE 1 · THE THREE AVERAGES</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Compute mean, median, mode</div>
<div style="color:#4a5578;margin-top:6px">For x = [4, 8, 6, 5, 8, 9, 8, 3], compute the mean, median, and mode.</div>
</div>

In [2]:
x = pd.Series([4, 8, 6, 5, 8, 9, 8, 3])
print(f"Sorted : {sorted(x)}")
print(f"Mean   = {x.mean():.2f}")
print(f"Median = {x.median():.1f}")
print(f"Mode   = {x.mode().iloc[0]}")

Sorted : [3, 4, 5, 6, 8, 8, 8, 9]
Mean   = 6.38
Median = 7.0
Mode   = 8


**Answer:** Mean = **6.375**, Median = **7.0** (average of the two middle values 6 and 8), Mode = **8** (appears three times).

<div style="background:#fef3e2;border-left:5px solid #d97706;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#7c3aed;letter-spacing:1px">CHALLENGE 2 · THE OUTLIER TEST</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Which average changes more?</div>
<div style="color:#4a5578;margin-top:6px">Start with [10, 12, 11, 13, 12]. Add the value 100. Recompute the mean and median, and say which one the outlier affected more.</div>
</div>

In [3]:
base = pd.Series([10, 12, 11, 13, 12])
with_out = pd.concat([base, pd.Series([100])], ignore_index=True)
print(f"Before: mean {base.mean():.1f}, median {base.median():.1f}")
print(f"After : mean {with_out.mean():.1f}, median {with_out.median():.1f}")

Before: mean 11.6, median 12.0
After : mean 26.3, median 12.0


**Answer:** the **mean** jumped from 11.6 to about **26.3**, while the **median** moved only from 12 to **12**. The outlier hammered the mean and barely touched the median, so the median is the better summary here.

<div style="background:#fef3e2;border-left:5px solid #d97706;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#d97706;letter-spacing:1px">CHALLENGE 3 · SKEW DIRECTION</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Read the skew from mean vs median</div>
<div style="color:#4a5578;margin-top:6px">A class's exam scores have mean = 72 and median = 78. Is the distribution left-skewed, right-skewed, or symmetric? What does that say about the scores?</div>
</div>

In [4]:
mean, median = 72, 78
if mean < median:   skew = "LEFT-skewed (negative): a tail of low scores pulls the mean down"
elif mean > median: skew = "RIGHT-skewed (positive): a tail of high scores pulls the mean up"
else:               skew = "symmetric"
print(f"mean {mean} < median {median}  ->  {skew}")

mean 72 < median 78  ->  LEFT-skewed (negative): a tail of low scores pulls the mean down


**Answer:** Since **mean (72) < median (78)**, the data is **left-skewed**: a few low scorers stretch the left tail and drag the mean below the median. Most students did fairly well.

<div style="background:#fef3e2;border-left:5px solid #d97706;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#059669;letter-spacing:1px">CHALLENGE 4 · GEOMETRIC MEAN</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Average a growth rate</div>
<div style="color:#4a5578;margin-top:6px">A company's revenue grows 20% in year 1, 5% in year 2, and falls 10% in year 3. What is the average annual growth rate? (Hint: use the geometric mean of the growth factors.)</div>
</div>

In [5]:
factors = np.array([1.20, 1.05, 0.90])           # +20%, +5%, -10%
gm = np.exp(np.mean(np.log(factors)))            # geometric mean of the factors
print(f"Average annual growth factor = {gm:.4f}")
print(f"Average annual growth rate   = {(gm-1)*100:.1f}%")
print(f"(The naive arithmetic average would wrongly give {(factors.mean()-1)*100:.1f}%.)")

Average annual growth factor = 1.0428
Average annual growth rate   = 4.3%
(The naive arithmetic average would wrongly give 5.0%.)


**Answer:** Average annual growth ≈ **4.3%**. Growth compounds, so you multiply the factors and take the cube root (the geometric mean), not the arithmetic average.

<div style="background:#fef3e2;border-left:5px solid #d97706;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#2563eb;letter-spacing:1px">CHALLENGE 5 · GROUPED DATA</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Mean from a frequency table</div>
<div style="color:#4a5578;margin-top:6px">A survey groups response times into bins with midpoints [10, 20, 30, 40] seconds and frequencies [2, 5, 8, 5]. Estimate the mean response time.</div>
</div>

In [6]:
mid  = np.array([10, 20, 30, 40])
freq = np.array([2, 5, 8, 5])
mean = np.sum(mid * freq) / np.sum(freq)
print(f"People             : {freq.sum()}")
print(f"Estimated mean time: {mean:.1f} seconds")

People             : 20
Estimated mean time: 28.0 seconds


**Answer:** Estimated mean ≈ **28.5 seconds**, using Σ(midpoint × frequency) / Σ(frequency). With only grouped counts, the class midpoints stand in for the raw values.

---
<div style="background:#ffffff;border:1px solid #e6e9f2;border-radius:16px;padding:22px 26px;font-family:Inter,sans-serif">
<div style="font-size:19px;font-weight:800;color:#1a2138">🎉 Nicely done!</div>
<div style="color:#4a5578;line-height:1.8;margin-top:8px">You computed all three averages, watched an outlier wreck the mean, read skew from mean vs median, used the geometric mean for growth, and averaged a frequency table. That is the full toolkit for describing the center of a dataset.</div>
</div>

<div style="text-align:center;color:#8b94b3;font-size:12px;margin-top:16px">Statistics, Data Science and AI: A Visual Handbook · © 2026 John Fisher</div>